### PR TITLE
Canary/data table column width functionality

### PR DESCRIPTION
**Pull request by jd239**
## Summary of the changes
Added props which allow the table to be resized as well as added functionality to retrieve column width / min / max width from the columns array.
After exploring the implementation of the native HTML table element, some of the sizing acceptance criteria require the table layout to be either `auto` or `fixed`. For example, it is not possible for the min/max column widths to work with a table-layout set to `fixed` and width `100%`. 
The onus has been placed on the developer to set the table-layout rather than the component automatically make that switch. 

Table layout: https://developer.mozilla.org/en-US/docs/Web/CSS/table-layout

To view the min/max column functionality, the table will need to be the following:
- width: `auto`.
- tableLayout `auto`

The default value for the table-layout will still be `fixed` even though this goes against the AC. This is because switching it to `auto` and removing the `100%` width will break VR tests for us and the consumer. 

Some functionality such as horizontal scrollbars require an additional wrapper with a width or min/max width. Please view the Cypress tests to see the example in code.

The row height functionality was added in a previous PR. 

## Related issue
#1005 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

N/A: Update only includes table and column sizing

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.stories.mdx [hunk 1]
```diff
@@ -1528,6 +1529,14 @@ const data = [
...
 ```
 
+### Sizing
+
+<Canvas withSource="none">
+  <Story name="Sizing" parameters={{ loki: { skip: true } }}>
+    {DataTableSizing()}
+  </Story>
+</Canvas>
+
```
Comment (GCHQ-Developer-530) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.stories.mdx:1539]: Can you add a code example for the sizing story? Same for canary-react as well

**Review by GCHQ-Developer-530 — DISMISSED** [review]

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx [hunk 2]
```diff
@@ -167,6 +173,16 @@ export class DataTable {
     showBackground?: boolean;
   };
 
+  /**
+   * Sets the maximum width of the data table
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx:177]: be good to specify unit supported eg px?, %? etc

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx [hunk 3]
```diff
@@ -167,6 +173,16 @@ export class DataTable {
...
+   */
+  @Prop() maxWidth?: string;
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx:179]: same re unit

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx [hunk 4]
```diff
@@ -167,6 +173,16 @@ export class DataTable {
...
+
+  /**
+   * Sets the minimum width of the data table
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx:182]: same re unit

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx [hunk 5]
```diff
@@ -222,10 +238,23 @@ export class DataTable {
...
   @Prop() stickyRowHeaders?: boolean = false;
+  /**
+   * Sets the layout of the table
+   */
+  @Prop() tableLayout?: "fixed" | "auto" = "fixed";
 
   /**
-   * Sets the method used to truncate longer text in cells where textWrap is `false`.
-   * The `tooltip` truncation pattern allows the overflowing text to be seen in a tooltip. The `show-hide` truncation pattern allows the overflowing text to be shown and hidden using the ic-typography "See more" / "See less" buttons.
+   * Sets the table width.
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx:247]: same re unit

File: packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx [hunk 6]
```diff
@@ -222,10 +238,23 @@ export class DataTable {
...
+   */
+  @Prop() width?: string;
+
+  /**
+   * Sets the table height.
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx:252]: same re unit

**Review by ad9242 — CHANGES_REQUESTED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**